code-challenges.test.js

```diff
@@ -34,8 +69,40 @@ const letterA = "a"
 const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+// RED
+
+describe("containsLetter", () => {
+    it("takes in an array of words and a letter, and returns an array of the provided words that contain the provided letter", () => {
+        expect(containsLetter(fruitArray, letterA)).toEqual(["Mango", "Apricot", "Peach"])
+        expect(containsLetter(fruitArray, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+
 // b) Create the function that makes the test pass.
 
+// GREEN
+// create a function that takes in an array of words, as well as a single letter, and returns a new array with the words containing the given letter
+// input: an array of words and one letter
+// output: a new array of the words that contain the given letter
+// first, filter over the given array so that a subset of that array is returned
+// use .includes() to search for the given letter within the array, and make this the condition the filter needs to meet to stop iterating
+// also, use .toUpperCase() to account for instances of the provided letter that are capitalized
+
+const containsLetter = (array, letter) => {
+    let newArr = array.filter(value => {
+        return value.toUpperCase().includes(letter.toUpperCase())
+    })
+    return newArr
+}
+console.log(containsLetter(fruitArray, letterA))
+// output: [ 'Mango', 'Apricot', 'Peach' ]
+console.log(containsLetter(fruitArray, letterE))
+// output: [ 'Cherry', 'Blueberry', 'Peach' ]
+
+// notes: 
+// The biggest struggle I had with this code was getting it to run without an error. My terminal was showing that I was getting the right output, but I continued to receive the error that 'array.filter' was not a function. Thanks to a classmate looking over my code, I realized I had forgotten to place fruitArray in my test.
```

Comment:
Fresh eyes are sometimes a blessing! I once struggled for 3 days on a bug that a cohort mate solved in 2 minutes max.